dotcom-rendering/src/components/Slideshow.tsx

```diff
@@ -1,61 +1,59 @@
 import { css, keyframes } from '@emotion/react';
-import { from, neutral, textSans, until } from '@guardian/source-foundations';
+import {
+	from,
+	palette,
+	space,
+	textSans,
+	until,
+} from '@guardian/source-foundations';
+import { takeFirst } from '../lib/tuple';
 import type { DCRSlideshowImage } from '../types/front';
 import type { ImageSizeType } from './Card/components/ImageWrapper';
 import { CardPicture } from './CardPicture';
 
-/**
- * It might look like you can change these and expect the animation to adapt accordingly
- * but it probably won't 😞
- */
-const HANG_TIME = 5;
-const FADE_TIME = 1;
+/** in seconds */
+const DISPLAY_DURATION = 5;
+const FADING_DURATION = DISPLAY_DURATION / 5;
 
 /**
  * We pass the result of the Emotion `keyframes` utility (decided here) to the animation-name
  * css property
  */
-function decideAnimationName({ slideshowLength }: { slideshowLength: number }) {
-	const totalLoopTime = slideshowLength * (HANG_TIME + FADE_TIME);
+function decideAnimationName(slideshowLength: number) {
+	if (slideshowLength === 1) {
+		return keyframes`from {opacity: 1} to {opacity:1}`;
+	}
+
+	const totalLoopTime = decideDuration(slideshowLength) / 100;
 	// Calculate the percentages for how long each animation stage lasts
-	const stageOne = `0%`;
-	const stageTwo = `${(100 / totalLoopTime) * FADE_TIME}%`;
-	const stageThree = `${(100 / totalLoopTime) * (FADE_TIME + HANG_TIME)}%`;
-	const stageFour = `${
-		(100 / totalLoopTime) * (FADE_TIME + HANG_TIME + FADE_TIME)
-	}%`;
+	const stages = [
+		0,
+		FADING_DURATION,
+		FADING_DURATION + DISPLAY_DURATION + FADING_DURATION,
+		FADING_DURATION + DISPLAY_DURATION + FADING_DURATION + FADING_DURATION,
+	] as const;
 	// Generate and return a keyframes name for this animation
 	return keyframes`
-		${stageOne} {opacity: 0;}
-		${stageTwo} {opacity: 1;}
-		${stageThree} {opacity: 1;}
-		${stageFour} {opacity: 0;}
+		${`${stages[0] / totalLoopTime}%`} {opacity: 0;}
+		${`${stages[1] / totalLoopTime}%`} {opacity: 1;}
+		${`${stages[2] / totalLoopTime}%`} {opacity: 1;}
+		${`${stages[3] / totalLoopTime}%`} {opacity: 0;}
```

Comment:
I don't think this is a necessary change, but it looks like there might be a coupling between the stage and the opacity so would it make sense to capture them together like this? Happy for you to tell me this doesn't make sense!

```ts
const stages = [
	{ time: 0, opacity: 0 },
	{ time: FADING_DURATION, opacity: 1 },
	{ time: FADING_DURATION + DISPLAY_DURATION + FADING_DURATION, opacity: 1 },
	{ time: FADING_DURATION + DISPLAY_DURATION + FADING_DURATION + FADING_DURATION, opacity: 0 },
];

return keyframes(stages.map(({ time, opacity }) => `${time / totalLoopTime}% { opacity: ${opacity}; }`));
```